src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -68,6 +70,116 @@ public function is_accessible(): bool {
 		return true;
 	}
 
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headers ) && $headers[0] !== 'HTTP/1.1 200 OK' ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the store sample product landing pages has redirects through 3P domains.
+	 *
+	 * @return bool
+	 */
+	public function has_redirects(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headesr ) && isset( $headers[0] ) ) {
+			if ( $headers[0] === 'HTTP/1.1 302 Found' ) {
+				// this is the URL where it's redirecting
+				return true;
+			}
+		}
+			return false;
+	}
+
+	/**
+	 * Get the headers of the store sample product landing pages.
+	 *
+	 * @return array
+	 */
+	public function get_landing_page_headers(): array {
+		$products = wc_get_products(
+			[
+				'limit' => 1,
+			]
+		);
+		if ( ! empty( $products ) ) {
+			$response = wp_remote_get( $products[0]->get_permalink() );
+			if ( is_array( $response ) && ! is_wp_error( $response ) ) {
+				return $response['headers'];
+			}
+		}
+		return [];
+	}
+
+	/**
+	 * Check if the merchant set the restrictions in robots.txt or not in the store.
+	 *
+	 * @return bool
+	 */
+	public function has_restriction(): bool {
+		return ! $this->robots_allowed( $this->get_site_url() );
+	}
+
+	/**
+	 * Check if the robots.txt has restrictions or not in the store.
+	 *
+	 * @param string $url
+	 * @return bool
+	 */
+	private function robots_allowed( $url ) {
+		$agents = [ preg_quote( '*', '/' ) ];
+		$agents = implode( '|', $agents );
+
+		// location of robots.txt file
+		$robotstxt = file( trailingslashit( $url ) . 'robots.txt' );
+
+		// if there isn't a robots, then we're allowed in
+		if ( empty( $robotstxt ) ) {
+			return true;
+		}
+
+		$rules        = [];
+		$rule_applies = false;
+		foreach ( $robotstxt as $line ) {
+			// skip blank lines
+			$line = trim( $line );
+			if ( ! $line ) {
+				continue;
+			}
+
+			// following rules only apply if User-agent matches '*'
+			if ( preg_match( '/^\s*User-agent:\s*(.*)/i', $line, $match ) ) {
+				$rule_applies = preg_match( "/($agents)/i", $match[1] );
+			}
+			if ( $rule_applies && preg_match( '/^\s*Disallow:\s*(.*)/i', $line, $regs ) ) {
+				// an empty rule implies full access - no further tests required
+				if ( ! $regs[1] ) {
+					return true;
+				}
+				// add rules that apply to array for testing
+				$rules[] = preg_quote( trim( $regs[1] ), '/' );
```

Comment:
Here again if all we are going to compare is the disallow of `/` then we don't need to add it to an array of rules and we can just compare if `'/' === trim( $regs[1] )`.

Although combined with matching a user agent of `*`, I'm wondering how useful this check will be. Because it will need to have a robots file with the following content to match:

```
User-agent: *
Disallow: /
```

Which would imply that they are restricting the full site.
Even if that pattern is not in the robots.txt file it's still possible to for example, restrict all product pages.

Reply:
Thanks for your suggestions! Based on this [PRD](https://docs.google.com/document/d/1NxMLzh5rb6KBx7xP7KZF2q4WeoAQJmnPTxE3B8Ss3jY/edit?resourcekey=0-QRep8wnQJgcCQRudP770GA#bookmark=id.km9dsgduk5uq), we could only do the general checking. Do you have any good suggestions?

Reply:
I don't have any suggestions for improving the checks. I was mainly pointing out that it will catch very few sites that don't match those patterns. Since these are only preliminary checks to get them through the onboarding and it will be checked again through an external service, it seems like this is as far as we can go for now.

Reply:
Thanks for your reply! I totally agree with you. This is only preliminary checks now.